src/main/java/com/example/bookshop/service/implementation/ShoppingCartImpl.java

```diff
@@ -0,0 +1,76 @@
+package com.example.bookshop.service.implementation;
+
+import com.example.bookshop.dto.AddBookShoppingCartRequestDto;
+import com.example.bookshop.dto.ShoppingCartDto;
+import com.example.bookshop.mapper.ShoppingCartMapper;
+import com.example.bookshop.model.Book;
+import com.example.bookshop.model.CartItem;
+import com.example.bookshop.model.User;
+import com.example.bookshop.repository.BookRepository;
+import com.example.bookshop.repository.ShoppingCartRepository;
+import com.example.bookshop.repository.UserRepository;
+import com.example.bookshop.service.ShoppingCartService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartImpl implements ShoppingCartService {
+    private ShoppingCartRepository shoppingCartRepository;
+    private ShoppingCartMapper shoppingCartMapper;
+    private BookRepository bookRepository;
+    private UserRepository userRepository;
+
+    @Override
+    public Page<ShoppingCartDto> findAll(Pageable pageable) {
+        return new PageImpl<>(shoppingCartRepository.findAll(pageable)
+                .stream()
+                .map(shoppingCartMapper::toDto)
+                .toList());
+    }
+
+    @Override
+    public void save(AddBookShoppingCartRequestDto addBookShoppingCartRequestDto,
+                                UserDetails userDetails) {
+        Book book = bookRepository.getReferenceById(addBookShoppingCartRequestDto.getBookId());
+        CartItem cartItem = new CartItem();
```

Comment:
find shop cart by id, check if cartItems in shop cart already include this book,  add book, save shop cart to db